File: desktop_computing/agent/jsontest.pl
#!/usr/bin/perl -w

use JSON;

my $utf8_encoded_json_text = '{   "1" : {      "stderr_file" : "OAR.1.stderr",      "stdout_file" : "OAR.1.stdout",      "directory" : "/root",      "command" : "date",      "state" : "toLaunch"   }}'
;

$perl_hash_or_arrayref = decode_json $utf8_encoded_json_text;

foreach $item (values %$perl_hash_or_arrayref) {
  print encode_json $item;
}

File: desktop_computing/agent/test/jsontest.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'json'

job_hash = JSON.parse('{   "1" : {      "stderr_file" : "OAR.1.stderr",      "stdout_file" : "OAR.1.stdout",      "directory" : "/root",      "command" : "date",      "state" : "toLaunch"   }}')

job_hash.each_key do |k|
  system job_hash[k]["command"]
end

puts 'working!'
